src/modules/users/application/services/auth.service.ts

```diff
@@ -0,0 +1,10 @@
+import { authConfig } from '@config/auth.config';
+import { IAuthService } from '@modules/users/domain/interfaces/auth-service.interface';
+import { IJWTPayload } from '@modules/users/domain/interfaces/jwt-payload.interface';
+import jwt from 'jsonwebtoken';
+
+export class AuthService implements IAuthService {
+	public signJWT(payload: IJWTPayload): string {
+		return jwt.sign(payload, authConfig.jwtSecret);
+	}
+}
```

Comment:
I have a doubt here, Do you think this should be a provider instead?

Reply:
The signJWT method could come from a provider, so we use it through an interface

Reply:
Done, moved to a provider